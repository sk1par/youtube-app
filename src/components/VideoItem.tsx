import { useDispatch } from 'react-redux';
import {history} from '../actions';
import selectedVideo from '../actions/selectedVideoActions';
import IVideos from '../models/videos.interface';

interface VideosProps {
    video: IVideos;
}

const VideoItem = ({video }: VideosProps) => {
    const dispatch = useDispatch();

    const selectVideo = (video: IVideos) => {
        dispatch(selectedVideo(video));
        dispatch(history(video));
    }

    return (
        <div onClick={() => selectVideo(video)} className='video-item item'>
            <img className='ui image' src={video.snippet.thumbnails.medium.url} alt={video.snippet.description}/>
            <div className='content'>
                <div className='header '>{video.snippet.title}</div>
            </div>
        </div>
    )
};
export default VideoItem;