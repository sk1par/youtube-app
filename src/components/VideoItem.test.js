import { useSelector } from 'react-redux';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';
import VideoItem from './VideoItem';
import selectedVideo from '../actions/selectedVideoActions';
import historyAction from '../actions/historyActions';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
    useDispatch: () => (fn) => () => fn(),
    useSelector: jest.fn(() => []),
}));

jest.mock('../actions/selectedVideoActions', () => {
    const selectedVideo = jest.fn();
    return selectedVideo;
});

jest.mock('../actions/historyActions', () => {
    const historyAction = jest.fn();
    return historyAction;
});

const getVideo = () => ({
    id: {
        videoId: '1',
    },
    snippet: {
        description: 'test',
        title: 'test',
        thumbnails: {
            medium: {
                url: 'test',
            },
        },
    },
});

it('should render single video div', () => {
    const wrapper = shallow(<VideoItem video={getVideo()} />);
    const title = wrapper.find('[data-auto-id="video-div"]');
    expect(title).toHaveLength(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should receive obj after click', () => {
    const v = getVideo();
    useSelector.mockReturnValueOnce([v]);
    const wrapper = shallow(<VideoItem video={v} />);
    const item = wrapper.find('[data-auto-id="video-div"]');
    item.simulate('click');
    expect(selectedVideo).toHaveBeenCalledTimes(1);
    expect(selectedVideo).toHaveBeenCalledWith(v);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should add video obj to history', () => {
    const v = getVideo();
    useSelector.mockReturnValueOnce([v]);
    const wrapper = shallow(<VideoItem video={v} />);
    const item = wrapper.find('[data-auto-id="video-div"]');
    item.simulate('click');
    expect(historyAction).toHaveBeenCalledTimes(1);
    expect(historyAction).toHaveBeenCalledWith(v);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});