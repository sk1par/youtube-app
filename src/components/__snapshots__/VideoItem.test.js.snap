// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add video obj to history 1`] = `
<div
  className="video-item item"
  data-auto-id="video-div"
  onClick={[Function]}
>
  <img
    alt="test"
    className="ui image"
    src="test"
  />
  <div
    className="content"
    data-auto-id="item-video"
  >
    <div
      className="header "
    >
      test
    </div>
  </div>
</div>
`;

exports[`should receive obj after click 1`] = `
<div
  className="video-item item"
  data-auto-id="video-div"
  onClick={[Function]}
>
  <img
    alt="test"
    className="ui image"
    src="test"
  />
  <div
    className="content"
    data-auto-id="item-video"
  >
    <div
      className="header "
    >
      test
    </div>
  </div>
</div>
`;

exports[`should render single video div 1`] = `
<div
  className="video-item item"
  data-auto-id="video-div"
  onClick={[Function]}
>
  <img
    alt="test"
    className="ui image"
    src="test"
  />
  <div
    className="content"
    data-auto-id="item-video"
  >
    <div
      className="header "
    >
      test
    </div>
  </div>
</div>
`;
